Use spdlog::spdlog_header_only

Switch to the header-only version of spdlog, since MacPorts currently ships
a precompiled spdlog.a that contains symbols compatible with fmt8, but we're
using fmt9.

Upstream-Status: Inappropriate [configuration]

--- source/libsys/CMakeLists.txt.orig	2022-08-13 14:41:41.000000000 +0200
+++ source/libsys/CMakeLists.txt	2022-08-13 14:42:11.000000000 +0200
@@ -16,7 +16,7 @@
         ${CMAKE_DL_LIBS}
         result_a
         fmt::fmt
-        spdlog::spdlog)
+        spdlog::spdlog_header_only)
 
 if (ENABLE_UNIT_TESTS)
     add_executable(sys_unit_test
--- source/intercept/CMakeLists.txt.orig	2022-08-13 14:41:41.000000000 +0200
+++ source/intercept/CMakeLists.txt	2022-08-13 14:42:04.000000000 +0200
@@ -51,7 +51,7 @@
         rpc_a
         sys_a
         result_a
-        spdlog::spdlog)
+        spdlog::spdlog_header_only)
 
 
 # Intercept orchestrator executable.
@@ -94,7 +94,7 @@
         result_a
         sys_a
         rpc_a
-        spdlog::spdlog)
+        spdlog::spdlog_header_only)
 
 
 # Intercept report sender executable.
--- source/bear/CMakeLists.txt.orig	2022-08-13 14:41:41.000000000 +0200
+++ source/bear/CMakeLists.txt	2022-08-13 14:42:25.000000000 +0200
@@ -7,7 +7,7 @@
         sys_a
         flags_a
         fmt::fmt
-        spdlog::spdlog)
+        spdlog::spdlog_header_only)
 
 include(GNUInstallDirs)
 install(TARGETS bear
--- source/citnames/CMakeLists.txt.orig	2022-08-13 14:41:41.000000000 +0200
+++ source/citnames/CMakeLists.txt	2022-08-13 14:42:20.000000000 +0200
@@ -12,7 +12,7 @@
         shell_a
         sys_a
         fmt::fmt
-        spdlog::spdlog
+        spdlog::spdlog_header_only
         nlohmann_json::nlohmann_json)
 target_compile_options(citnames_json_a PRIVATE -fexceptions)
 
@@ -44,7 +44,7 @@
         sys_a
         exec_a
         fmt::fmt
-        spdlog::spdlog)
+        spdlog::spdlog_header_only)
 
 # Create an executable from the sub projects.
 add_executable(citnames
--- source/libmain/CMakeLists.txt.orig	2022-08-13 14:41:41.000000000 +0200
+++ source/libmain/CMakeLists.txt	2022-08-13 14:42:30.000000000 +0200
@@ -11,4 +11,4 @@
         result_a
         flags_a
         fmt::fmt
-        spdlog::spdlog)
+        spdlog::spdlog_header_only)
